src/construct.ts

```diff
@@ -536,3 +551,10 @@ export interface MetadataOptions {
    */
   readonly traceFromFunction?: any;
 }
+
+// Mark all instances of 'Construct'
+Object.defineProperty(Construct.prototype, CONSTRUCT_SYM, {
+  value: true,
+  enumerable: false,
+  writable: false,
+});
```

Comment:
The difference is where the symbol lives: on the *instance* or on the *class*.

This puts it on the class, putting it in the construct (and presumably passing `this` instead of `Construct.prototype`) would put it on the instance.

Putting it on the class saves an instruction on every construct instantiation, and the memory of saving it in every symbol table (plus it might potentially have some impact on V8s optimizations... though I'm for sure not looking at any of that!)

It's probably a minor thing, but this ought to be slightly more efficient.